src/fides/api/util/connection_util.py

```diff
@@ -208,6 +206,9 @@ def patch_connection_configs(
                 if isinstance(value, bool) or value
             }
 
+        if system:
```

Comment:
This needed to be moved down. The if statement above was completely recreating the `config_dict` on edits causing the `system.id` assignment to be removed.